components/candle/challengeService.ts

```diff
@@ -1140,10 +1143,14 @@ export class ChallengeService extends ChallengeRegistry {
         //NOTE: Official will always grant XP to both Location Mastery and the Player Profile
         const actionXp = challenge.Xp || 0
         const masteryXp = challenge.Rewards?.MasteryXP || 0
```

Comment:
Kind of in doubt here, but I would move any mention of XP and drops to the progressionService as well... It's kind of nice how `grantProfileProgression` can be called without a challenge.

Either way, please move the `//NOTE` as it's more applicable above the mentioned `const xp`.